server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1860,9 +1860,14 @@ private CompactionStats _majorCompact(MajorCompactionReason reason) throws IOExc
 
       if (inputFiles.isEmpty()) {
         if (reason == MajorCompactionReason.USER) {
-          // no work to do
-          lastCompactID = compactionId.getFirst();
-          updateCompactionID = true;
+          if (compactionId.getSecond().getIterators().isEmpty()) {
+            log.debug("No-op major compaction by USER on 0 input files because no iterators present.");
+            lastCompactID = compactionId.getFirst();
+            updateCompactionID = true;
+          } else {
+            log.debug("Major compaction by USER on 0 input files with iterators.");
+            filesToCompact = new HashMap<FileRef,DataFileValue>();
```

Comment:
Yup, it looks like we moved this fixVersion to 1.7.0 which is on Java7. Accumulo 1.6 and earlier is still on java6